### ISSUE TITLE
Helping out with maintanence?

### ISSUE THREAD

**Issue by Lyude** [daa84/neovim-gtk#267]
Hi! I'm a long time neovim-gtk user, and for a while now I've actually been working on making quite a number of improvements to neovim-gtk on [a fork I've been maintaining](https://github.com/lyude/neovim-gtk/tree/wip/nvim-rs) on my Github. Some of the changes here I've made are pretty significant, [like the conversion of neovim-gtk from neovim-lib to nvim-rs](https://github.com/Lyude/neovim-gtk/commit/ca5d183887e4e353c64993036a64077f7f5bde62) so that we can use async code in neovim-gtk. I was thinking of just forking the project, but I wanted to see if you might be interested in integrating the changes I've made so far first. I'm also planning to work on moving neovim-gtk over to the latest Gtk+ version, and a coworker of mine might also be working on it in their spare time. Since I use neovim-gtk as my daily driver for work and don't really plan on ever changing my setup, I think I'm willing to help maintain neovim-gtk in the long term (granted, at my own pace based on how much spare time I have). I figured that might help since it seems like the project has been inactive for a little while now :).

FWIW as well if you're curious about my background, most of my experience with development is working on projects like the linux kernel ("Lyude Paul" is usually the git name I use there), and I've also contributed to projects like libinput, panfrost, Xorg, igt-gpu-tools, and a number of others.

**Comment by bellini666** [comment]
Hey @Lyude ,

Nice work there! I'm going to test you fork and use it as my daily driver if it works nice, was always with the impression that this was an "abandoned" project, but it is still the best neovim's UI for me (as someone who came from gvim for vim).

Would be awesome to join forces and merge your fork into here, but maybe the maintainer is not active anymore (considering the latest commit date and lack of replies to issues/PRs, including this one), so we might need to consider your fork the main one for contributions?

Also, would be nice to add it to this list, so it can attract more contributors: https://github.com/rockerBOO/awesome-neovim

**Comment by Lyude** [comment]
> Hey @Lyude ,
> 
> Nice work there! I'm going to test you fork and use it as my daily driver if it works nice, was always with the impression that this was an "abandoned" project, but it is still the best neovim's UI for me (as someone who came from gvim for vim).
> 
> Would be awesome to join forces and merge your fork into here, but maybe the maintainer is not active anymore (considering the latest commit date and lack of replies to issues/PRs, including this one), so we might need to consider your fork the main one for contributions?

Yep! That's my current plan, I figured I'd give the maintainer a bit of time at least before making the decision final on my end, as it'd be a shame to have to use a different name for this then neovim-gtk. Although I do like the name I'm currently considering going with, hdiq ("how do I quit?")

> 
> Also, would be nice to add it to this list, so it can attract more contributors: https://github.com/rockerBOO/awesome-neovim

**Comment by Lyude** [comment]
Oops, forgot to answer the second question - yeah, I will try to remember to add my project to the awesome-neovim list once I do the name change (probably the next time I sit down to resume work on the gtk+4 conversion)

**Comment by Lyude** [comment]
Update, since according to my history I finally have an audience! Will probably try to decide on a name I like this weekend (I'm wondering now if maybe I should just name it nvimg (gnvim is taken), and then create a new repo and move things from my fork over there. Until then, I moved work over to [this branch](https://github.com/Lyude/neovim-gtk/tree/main).

**Comment by bellini666** [comment]
Hey @Lyude , awesome! 

I already started using your fork, looking forward to new features being added to it! Although I have some good experience with gtk using python, I lack experience with rust... I'll try to help there if I can as neovim-gtk (now on your fork) became my daily driver!

Will at least try to provide some feedback and insights on any issues that I find!

**Comment by jacobmischka** [comment]
> ... and then create a new repo and move things from my fork over there.

Sorry, why create a new repo? I don't see a problem with showing that the project is a fork, it seems disingenuous to not do that in my opinion.

I tested your fork out a bit the other day, though ran into issues with configuration flags not working correctly (to disable popup and tabline). I plan on revisiting at some point and creating some PRs with my own fixes (mostly the outstanding ones in this repo), though I was hoping those flags might be resolved on their own once you finish your ongoing updates.

Thanks again for picking this project up!

**Comment by diefans** [comment]
@Lyude eventually it makes sense to enable the issue tracker on your fork, so people can properly address them...

**Comment by bellini666** [comment]
> @Lyude eventually it makes sense to enable the issue tracker on your fork, so people can properly address them...

Yeah, was going to say that. I already have an issue to report: Tryint to disable the popup menu (https://github.com/daa84/neovim-gtk/wiki/Configuration#options) with `call rpcnotify(1, 'Gui', 'Option', 'Popupmenu', 0)` is not working in your fork.

**Comment by Lyude** [comment]
> > ... and then create a new repo and move things from my fork over there.
> 
> Sorry, why create a new repo? I don't see a problem with showing that the project is a fork, it seems disingenuous to not do that in my opinion.

I mean, for one I have every intent of crediting folks involved (probably on the README.md at the top so it's easily visible when you find the project), crediting folks for their work is important. Regardless though it's not being a fork I have an issue with, moreso as I was under the impression it was more difficult to discover forks of repos then it was original repos? This is just from my memory though, and the vast majority of development I do is outside of Github (usually Gitlab; which I prefer quite a bit for it's code review tools; or mailing lists) so I could be 100% wrong about this. If being a fork doesn't actually make it any harder for people to google "GTK neovim client" and find this project then I'm totally happy to keep it as a fork.

> 
> I tested your fork out a bit the other day, though ran into issues with configuration flags not working correctly (to disable popup and tabline). I plan on revisiting at some point and creating some PRs with my own fixes (mostly the outstanding ones in this repo), though I was hoping those flags might be resolved on their own once you finish your ongoing updates.

Yeah that sounds like a blindspot in my testing :)

> 
> Thanks again for picking this project up!

no problem! I'm glad to see other folks have interest in this, as I'd definitely like to eventually have more maintainers once more issues start coming in (of course, contributions are welcome at any point! just beware there's still a number of large changes I'm planning on making). Also, I will make sure I've got the issues option on my repo enabled as some folks have pointed out.

**Comment by Lyude** [comment]
> > @Lyude eventually it makes sense to enable the issue tracker on your fork, so people can properly address them...
> 
> Yeah, was going to say that. I already have an issue to report: Tryint to disable the popup menu (https://github.com/daa84/neovim-gtk/wiki/Configuration#options) with `call rpcnotify(1, 'Gui', 'Option', 'Popupmenu', 0)` is not working in your fork.

@jacobmischka (I forgot to mention you in the commit message whoops) the issue you and @bellini666 reported is fixed in lyude/neovim-gtk@4e466e34c058b0c24adb2e5c96e0b574b10c7de3
